File: generated-code/codecraft/go/model/attack_properties.go
package model

import "io"
import . "trans_gen_test/stream"

type AttackProperties struct {
    AttackRange int32
    Damage int32
    CollectResource bool
}

func NewAttackProperties(attackRange int32, damage int32, collectResource bool) AttackProperties {
    return AttackProperties {
        AttackRange: attackRange,
        Damage: damage,
        CollectResource: collectResource,
    }
}

func ReadAttackProperties(reader io.Reader) AttackProperties {
    var attackRange int32
    attackRange = ReadInt32(reader)
    var damage int32
    damage = ReadInt32(reader)
    var collectResource bool
    collectResource = ReadBool(reader)
    return AttackProperties {
        AttackRange: attackRange,
        Damage: damage,
        CollectResource: collectResource,
    }
}

func (attackProperties AttackProperties) Write(writer io.Writer) {
    attackRange := attackProperties.AttackRange
    WriteInt32(writer, attackRange)
    damage := attackProperties.Damage
    WriteInt32(writer, damage)
    collectResource := attackProperties.CollectResource
    WriteBool(writer, collectResource)
}

File: generated-code/codecraft/kotlin/src/main/kotlin/model/AttackProperties.kt
package model

import util.StreamUtil

class AttackProperties {
    var attackRange: Int = 0
    var damage: Int = 0
    var collectResource: Boolean = false

    constructor(attackRange: Int, damage: Int, collectResource: Boolean) {
        this.attackRange = attackRange
        this.damage = damage
        this.collectResource = collectResource
    }

    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, attackRange)
        StreamUtil.writeInt(stream, damage)
        StreamUtil.writeBoolean(stream, collectResource)
    }

    override fun toString(): String {
        var stringBuilder = StringBuilder("AttackProperties { ")
        stringBuilder.append("attackRange: ")
        stringBuilder.append(attackRange)
        stringBuilder.append(", ")
        stringBuilder.append("damage: ")
        stringBuilder.append(damage)
        stringBuilder.append(", ")
        stringBuilder.append("collectResource: ")
        stringBuilder.append(collectResource)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): AttackProperties {
            var attackRange: Int
            attackRange = StreamUtil.readInt(stream)
            var damage: Int
            damage = StreamUtil.readInt(stream)
            var collectResource: Boolean
            collectResource = StreamUtil.readBoolean(stream)
            return AttackProperties(attackRange, damage, collectResource)
        }
    }
}